src/hotspot/share/runtime/deoptimization.cpp

```diff
@@ -976,7 +976,7 @@ class BooleanBoxCache : public BoxCacheBase<java_lang_Boolean> {
 protected:
   static BooleanBoxCache *_singleton;
   BooleanBoxCache(Thread *thread) {
-    InstanceKlass* ik = find_cache_klass(java_lang_Boolean::symbol());
+    InstanceKlass* ik = find_cache_klass(java_lang_Boolean::symbol(), thread);
```

Comment:
If this throws a Java exception, you have to handle it here, you can't ignore it with thread.  You should initialize these classes during initialization.  You shouldn't be doing this in deoptimization.

Reply:
Yes. I changed my patch. Thank you for your review.